components/overlayTrigger/overlayTrigger.js

```diff
@@ -0,0 +1,155 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+export default class OverlayTrigger extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      active: false,
+    };
+  }
+
+  linkChild = (ref) => {
+    this.elem = ref;
+  }
+
+  componentDidMount() {
+    if (this.props.triggerAction !== 'click') {
+      return;
+    }
+
+    this.addOutsideClickListener();
+  }
+
+  componentWillUnmount() {
+    if (this.props.triggerAction !== 'click') {
+      return;
+    }
+
+    this.removeOutsideClickListener();
+  }
+
+  addOutsideClickListener() {
+    document.body.addEventListener('click', this.handleOutsideClick);
+  }
+
+  removeOutsideClickListener() {
+    document.body.removeEventListener('click', this.handleOutsideClick);
+  }
+
+  handleOutsideClick = (e) => {
+    if (this.elem.contains(e.target)) {
+      return;
+    }
+
+    this.state.active && this.hideElement();
+  }
+
+  toggleElement = () => {
+    this.state.active
+      ? this.hideElement()
+      : this.showElement();
+  }
+
+  showElement() {
+    const { onElementShow } = this.props;
+
+    this.setState({ active: true });
+
+    onElementShow && onElementShow();
+  }
+
+  hideElement() {
+    const { onElementHide } = this.props;
+
+    this.setState({ active: false });
+
+    onElementHide && onElementHide();
```

Comment:
Why not adding a:

```js

static defaultProps = {
  onElementHide = () => {},
  onElementShow = () => {},
}

```

This would allow you to do:

```onElementHide()```

(same for show)

Reply:
done